keep-ui/app/incidents/[id]/incident-workflow-sidebar.tsx

```diff
@@ -0,0 +1,176 @@
+import { Fragment, useEffect } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { Text, Button, TextInput, Badge, Title, Card } from "@tremor/react";
+import { IoMdClose } from "react-icons/io";
+import { WorkflowExecution } from "app/workflows/builder/types";
+import { formatDistanceToNowStrict } from "date-fns";
+import {
+  getIcon,
+  getTriggerIcon,
+  extractTriggerValue,
+} from "app/workflows/[workflow_id]/workflow-execution-table";
+import { useWorkflowExecution } from "utils/hooks/useWorkflowExecutions";
+
+interface IncidentWorkflowSidebarProps {
+  isOpen: boolean;
+  toggle: VoidFunction;
+  selectedExecution: WorkflowExecution | null;
+}
+
+const IncidentWorkflowSidebar: React.FC<IncidentWorkflowSidebarProps> = ({
+  isOpen,
+  toggle,
+  selectedExecution,
+}) => {
+  const { data: workflowExecutionData } = useWorkflowExecution(
+    selectedExecution?.workflow_id || "",
+    selectedExecution?.id || ""
+  );
+
+  useEffect(() => {
+    if (isOpen && selectedExecution) {
```

Comment:
Is it intended? If this is a comment for future, let's comment the whole useEffect to not to run it idle